inc/ThirdParty/Themes/SubscriberFactory.php

```diff
@@ -0,0 +1,120 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\ThirdParty\Themes;
+
+use ProxyManager\Factory\LazyLoadingValueHolderFactory;
+use ProxyManager\Proxy\LazyLoadingInterface;
+use WP_Rocket\ThirdParty\SubscriberFactoryInterface;
+
+class SubscriberFactory implements SubscriberFactoryInterface {
+	/**
+	 * Get a theme subscriber data
+	 *
+	 * @return array
+	 */
+	public function get_subscriber() {
+		$theme = ThemeResolver::get_current_theme();
+
+		switch ( $theme ) {
+			case 'avada':
+				return [
+					'class'     => Avada::class,
+					'arguments' => [
+						'options',
+					],
+				];
+			case 'bridge':
+				return [
+					'class'     => Bridge::class,
+					'arguments' => [
+						'options',
+					],
+				];
+			case 'divi':
+				return [
+					'class'     => $this->lazyload(
+						Divi::class,
+						[
+							'options_api',
+							'options',
+							'delay_js_html',
+							'rucss_used_css_controller',
+						]
+					),
+					'arguments' => [],
+				];
+			case 'flatsome':
+				return [
+					'class'     => Flatsome::class,
+					'arguments' => [],
+				];
+			case 'jevelin':
+				return [
+					'class'     => Jevelin::class,
+					'arguments' => [],
+				];
+			case 'minimalist_blogger':
+				return [
+					'class'     => MinimalistBlogger::class,
+					'arguments' => [],
+				];
+			case 'polygon':
+				return [
+					'class'     => Polygon::class,
+					'arguments' => [],
+				];
+			case 'uncode':
+				return [
+					'class'     => Uncode::class,
+					'arguments' => [],
+				];
+			case 'xstore':
+				return [
+					'class'     => Xstore::class,
+					'arguments' => [],
+				];
+			case 'themify':
+				return [
+					'class'     => Themify::class,
+					'arguments' => [
+						'options',
+					],
+				];
+			case 'shoptimizer':
+				return [
+					'class'     => Shoptimizer::class,
+					'arguments' => [],
+				];
+			default:
+				return [];
+		}
+	}
+
+	/**
+	 * Lazyload the subscriber
+	 *
+	 * @param string $class_name Class name.
+	 * @param array  $args Class constructor arguments.
+	 *
+	 * @return object
+	 */
+	private function lazyload( $class_name, $args ) {
+		$factory = new LazyLoadingValueHolderFactory();
+		$params  = [];
+
+		foreach ( $args as $arg ) {
+			$params[] = $this->getContainer()->get( $arg );
+		}
+
+		return $factory->createProxy(
+			$class_name,
+			function ( &$wrapped_object, LazyLoadingInterface $proxy, $method, array $parameters, &$initializer ) use ( $class_name, $params ) {
+				$initializer    = null; // disable initialization.
+				$wrapped_object = new $class_name( ...$params );
+
+				return true; // confirm that initialization occurred correctly.
+			}
+		);
+	}
```

Comment:
It was agreed with @MathieuLamiot to incorporate lazyload for the Divi subscriber in this PR

Reply:
> For the RUCSS controller we could just [share it](https://github.com/wp-media/wp-rocket/blob/1558300abaec2db586e29442b4d4657e9c12f94f/inc/Engine/Optimization/RUCSS/ServiceProvider.php#L97) instead of using a proxy and we could do the same thing for [the delay js class](https://github.com/wp-media/wp-rocket/blob/1558300abaec2db586e29442b4d4657e9c12f94f/inc/Engine/Optimization/DelayJS/ServiceProvider.php#L50).

I'll let @Tabrisrp confirm but my understanding is that, in the long run, we should remove dependencies on RUCSS controller (and maybe Delay JS too?) by using filters. Sharing them was discussed already (with Ahmed if I remember correctly) but not suitable for other use cases. So for now, we work around those heavy dependencies with the lazyload.

Reply:
After discussing with Cyrille, we agreed that this solution is not very good and the benefit vs cons is not worth it.

So I removed it, and focusing on improving the Divi class itself in a future update.